docs/adrs/003-Retry_endpoint_design.md

```diff
@@ -0,0 +1,162 @@
+* The retry will only continue if the status of the job to be retried is in [Status.terminated.value, Status.error.value]
+* If the job id points to a job that has already been retried, it will attempt to retry that job's `retry_parent` instead.
+* If the job id has never been retried, it becomes the `retry_parent` 
+* EE2 looks up the method versions and parameters, and then submits the job to be retried, incrementing the `retry_count` of the job being retried, and the newly launched job gains a field called `retry_parent` that contains the job id of the job from the original request.
+* The job is submitted and upon successful submission, the child job adds the field `retry_parent` and notifies the `parent_job_id` that a new `child_job` has been added by appending itself to the `parent_job.child_jobs[]` field
+* There is no way to specify ResourceRequirements with a retry at the moment, even if the job was previously submitted by an administrator and had specfified ResourceRequirements. The retry will only use resource requirements from the catalog / ee2 config.
+
+
+### Batch Behavior
+* If a job has the attribute of `batch_job=True` the retry will fail, since there is no method to re-run. This is a bug, as it doesn't fail gracefully.
+* If a job has the attribute of `batch_job=True`, but is actually a child job, the parent will be notified of this new retried job
+* Multiple in-flight retries are allowed.
+* Adds `child_job_id` to `parent_job_id.child_job_ids[]`
+
+## Retry_job behavior
+* Blocking and single submit to HTCondor. It should be fine as it returns relatively quickly
+  
+## Retry_jobs behavior
+* Submitting multiple jobs for retry serially calls the same code path
+used for running a single job and blocks until all jobs have been
+submitted to the condor queue. This can cause issues if the
+network drops, and makes the narrative not aware of the state of
+the retry. Submitting 100 jobs currently takes 9 seconds, and that
+is a lot of time for things to go wrong. 
+* (Follow up: Hopefully the making the narrative aware of the state of the retry will be mitigated by the narrative backend. It just blocks on the call anyway, with the default service timeout, which I think is something wacky like half an hour. As long as the user doesn't kill the kernel at that time, all should be well. Of course, if it were me, and it looked frozen for more than a couple minutes, I'd probably restart. )
+* Multiple in-flight retries are allowed.
+
+### Desired Behavior
+
+#### General
+* Prevent multiple in-flight retries to prevent the user from wasting their own resources (and the queues resources)
+* Non blocking job submission for submitting multiple jobs, possibly via using `run_job_batch` (requires refactor of run_job_batch)
+* One single submission to HTCondor instead of multiple job submissions
+* Ability to gracefully handle jobs with children
```

Comment:
I'm not totally sure what that means, can you elaborate?

Reply:
It used to throw an error about a method not found instead of "Cannot retry a batch parent" job.

Reply:
Fair enough, can you document what gracefully handling jobs with children means in the ADR

Reply:
Ok added.

Reply:
:+1: